clang/lib/Sema/SemaType.cpp

```diff
@@ -1494,7 +1494,8 @@ static QualType ConvertDeclSpecToType(TypeProcessingState &state) {
       Result = Context.DoubleTy;
     break;
   case DeclSpec::TST_float128:
-    if (!S.Context.getTargetInfo().hasFloat128Type() &&
+    if (!S.Context.getTargetInfo().hasFloat128Type() && 
+        !S.getLangOpts().SYCLIsDevice &&
```

Comment:
Because of this change we silent usage of `__float128` in device code.
Interesting observation: `__float128` type variable in device code turns into `fp128` type variable in LLVM IR and after conversion to SPIRV and back, I see `half` (!) type variable. 
@bader , @AlexeySachkov FYI.

Reply:
> Because of this change we silent usage of `__float128` in device code.

It sounds like a bug. We should not allow `__float128` in device code, so I guess the right fix is use differed diagnostic instead of skipping it.